packages/astro/src/@types/astro.ts

```diff
@@ -1772,8 +1772,8 @@ export interface EndpointOutput {
 	encoding?: BufferEncoding;
 }
 
-export type APIRoute = (
-	context: APIContext
+export type APIRoute<RouteProps extends Props = Props> = (
+	context: APIContext<RouteProps>
```

Comment:
There's a little breaking change with this, as `Props` is `Record<string, unknown>` and previously the default type is `Record<string, any>`. I think we should have something like this instead:

```suggestion
export type APIRoute<Props extends Record<string, any> = Record<string, any>> = (
	context: APIContext<Props>
```

Reply:
Ah didn't notice that. Moving the `export type Props = Record<string, unknown>;` after the `APIRoute` type should fix it.

Reply:
Done in 61af58e